frontend/src/views/Game.tsx

```diff
@@ -1,45 +1,65 @@
 import React, { useEffect, useState } from 'react';
-import { useLocation } from 'react-router-dom';
+import { useHistory, useLocation } from 'react-router-dom';
 import SplitterLayout from 'react-splitter-layout';
+import { useBeforeunload } from 'react-beforeunload';
 import Editor from '../components/core/Editor';
-import { ErrorResponse } from '../api/Error';
+import { errorHandler, ErrorResponse } from '../api/Error';
 import { Problem, getProblems } from '../api/Problem';
 import { Room } from '../api/Room';
-import {
+import MainContainer, {
   FlexContainer, FlexInfoBar, Panel, SplitterContainer,
 } from '../components/core/Container';
 import ErrorMessage from '../components/core/Error';
 import { ProblemHeaderText, Text } from '../components/core/Text';
 import 'react-splitter-layout/lib/index.css';
 import { checkLocationState } from '../util/Utility';
+import Loading from '../components/core/Loading';
 
 type LocationState = {
   room: Room,
 }
 
 function GamePage() {
+  // Get history object to be able to move between different pages
+  const history = useHistory();
   const location = useLocation<LocationState>();
   const [room, setRoom] = useState<Room | null>(null);
   const [problems, setProblems] = useState<Problem[]>([]);
+  const [fullPageLoad, setFullPageLoad] = useState<boolean>(true);
```

Comment:
Nit: naming this `fullPageLoading` may be more consistent since we have a `loading` state in Lobby.tsx

Reply:
Done!